src/ctapipe/io/interpolation.py

```diff
@@ -228,12 +230,12 @@ def add_table(self, tel_id, input_table):
         self._interpolators[tel_id]["alt"] = interp1d(mjd, alt, **self.interp_options)
 
 
-class CalibrationInterpolator(Interpolator):
+class GainInterpolator(Interpolator):
```

Comment:
I don't like the name GainInterpolator... There are many "gains" we deal with, and the names are often used incorrectly as "any multiplicative factor":
-  the HV gain setting (the only real "gain")
-  the "gain channel" (high or low)
- the DC/PE conversion factor (sometimes also called relative gain correction, which ensures a PE in one camera is a PE in another)
- the inter-pixel flat-field correction (which ensures the pixel response is flat across the camera as measured using a uniform light source).  

If this class is interpolating the flat-fielding values, please call it `FlatFieldInterpolator` to avoid confusion.   We also may in the future want to interpolate these other gain-correction values.  That also means that the name in the ctapipe output file should not be `gain` either, but `/dl1/calibration/flatfield/tel_x`.